cvpy/tests/test_cvat_project.py

```diff
@@ -0,0 +1,63 @@
+import sys
+import unittest
+
+import swat
+import xmlrunner
+
+from cvpy.annotation.base.AnnotationLabel import AnnotationLabel
+from cvpy.annotation.base.AnnotationType import AnnotationType
+from cvpy.annotation.base.Credentials import Credentials
+from cvpy.annotation.cvat.CVATProject import CVATProject
+
+
+class TestCVATProject(unittest.TestCase):
+
+    def test_cvat_project(self):
+        url = TestCVATProject.cvat_url
+
+        cas_session = swat.CAS(TestCVATProject.cas_host, TestCVATProject.cas_port)
+
+        project_name = 'Test Project'
+        annotation_type = AnnotationType.OBJECT_DETECTION
+
+        mountain_label = AnnotationLabel(name='Mountain', color='orange')
+        person_label = AnnotationLabel(name='Person', color='green')
+
+        labels = [mountain_label, person_label]
+
+        credentials = Credentials()
+        cvat_project = CVATProject(url=url, cas_session=cas_session, credentials=credentials,
+                                   project_name=project_name, annotation_type=annotation_type,
+                                   labels=labels)
+
+        self.assertEqual(cvat_project.url, url)
+        self.assertEqual(cvat_project.cas_session, cas_session)
+        self.assertEqual(cvat_project.credentials, credentials)
+        self.assertEqual(cvat_project.project_name, project_name)
+        self.assertEqual(cvat_project.annotation_type, annotation_type)
+        self.assertEqual(cvat_project.labels, labels)
+
+        self.assertIsNotNone(cvat_project.project_id)
+
+        # Delete the project from CVAT
+        cvat_project._delete_project_in_cvat()
+
+    def test_cvat_project_invalid_user_password(self):
+        url = TestCVATProject.cvat_url
+        credentials = Credentials(user='foo', password='bar')
+        try:
+            cvat_project = CVATProject(url=url, credentials=credentials)
+        except Exception as e:
+            self.assertEqual(str(e), "Unable to authenticate: ['Unable to log in with provided credentials.']")
+
+
+if __name__ == '__main__':
+    if len(sys.argv) > 1:
+        TestCVATProject.cvat_url = sys.argv.pop()
+        TestCVATProject.cas_host = sys.argv.pop()
+        TestCVATProject.cas_port = sys.argv.pop()
+
+    unittest.main(
+        testRunner=xmlrunner.XMLTestRunner(output='test-reports'),
+        failfast=False, buffer=False, catchbreak=False
+    )
```

Comment:
Moving forward, will these tests be run automatically against our team-wide CVAT server cvdata.unx.sas.com as we push new features?

Reply:
The CVAT url is an argument to the test, which we will set to the team-wide CVAT server (for now). So all unit tests even for new features will run against this server.